aten/src/ATen/native/mps/operations/TriangularOps.mm

```diff
@@ -172,5 +172,82 @@
 
 }
 
+Tensor trace_mps_out(const Tensor& self) {
+ // TODO: perform input checks here 
+
+  IntArrayRef input_size = self.sizes();
+  auto num_input_dims = input_size.size();
+  TORCH_CHECK(num_input_dims == 2, "trace_mps_out: Input tensor must be 2D");
+
+  using namespace mps;
+  MPSStream* stream = getCurrentMPSStream();
+
+  // Derive from MPSCachedGraph
+  struct CachedGraph : public MPSCachedGraph
+  {
+    CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor *inputTensor_ = nil;
+    MPSGraphTensor *outputTensor_ = nil;
+  };
```

Comment:
This is not needed, use https://github.com/pytorch/pytorch/blob/2e48b478e06b38a7468832d980d214441855547e/aten/src/ATen/native/mps/OperationUtils.h#L139-L144
```suggestion
```